src/Workspaces/Remote/ServiceHub/Shared/Extensions.cs

```diff
@@ -164,18 +164,28 @@ internal static partial class Extensions
             }
         }
 
-        private static bool ReportUnlessCanceled(Exception ex, CancellationToken token)
+#pragma warning disable CA1068 // this method accepts 2 cancellation tokens
```

Comment:
Can you add some comments here about the purpose of both the cancellation tokens?